dagshub/data_engine/datasources.py

```diff
@@ -30,18 +30,19 @@ def create_datasource(repo: str, name: str, path: str, revision: Optional[str] =
 
     parsed = urllib.parse.urlparse(path)
 
-    if parsed.scheme == "" \
-        and parsed.hostname == "" \
-        and parsed.query == "" \
-        and parsed.params == "" \
-        and parsed.fragment == "":
+    if parsed.path \
+       and not parsed.scheme \
+       and not parsed.hostname \
+       and not parsed.query \
+       and not parsed.params \
+       and not parsed.fragment:
         return create_from_repo(repo, name, path=parsed.path, revision=revision)
 
-    elif parsed.scheme != "" and parsed.hostname != "":
+    elif parsed.scheme and parsed.hostname:
         # Bucket URL
-        if parsed.query != "" or parsed.params != "" or parsed.fragment != "":
+        if parsed.query or parsed.params or parsed.fragment:
             raise ValueError("Invalid bucket URL: ", path)
-        if revision != "":
+        if revision:
```

Comment:
I checked what `urllib.parse` is returning, and it's returning empty strings on all fields. No Nones anywhere.
So that means that this one line is the actual line that broke stuff. I'd just check that the revision is not set, everything else is basically user error (although maybe it's a good idea for _us_ to check that a branch even exist)

```suggestion
        if revision is not None:
```

Reply:
For me it returned `None`s on some fields and it caused the whole function to.... malfunction ¯\_(ツ)_/¯

Reply:
The hostname, at least, is also `None`. The whole thing seems scarily inconsistent.
![image](https://github.com/DagsHub/client/assets/611655/86988324-6e19-40de-a8da-21157c46ccc2)

Reply:
Also, I don't want to check `revision is not None` - if it's an empty string, I feel comfortable just throwing it out

Reply:
> The hostname, at least, is also `None`. The whole thing seems scarily inconsistent.
Yea, that whole part of the core library is very weird IMO. Especially after how well paths work now, urls feel really bad now